rust/lance-index/src/vector/hnsw.rs

```diff
@@ -333,6 +368,49 @@ impl HNSW {
         writer.finish_with_metadata(&metadata).await?;
         Ok(())
     }
+
+    // Write partitioned HNSWs to the file.
+    pub async fn write_parted_hnsw(
```

Comment:
Arent these copied from ivf?

Reply:
they are, will modify this later in next PR. we need additional metadata to track all partitions